pkg/proxy/endpoints.go

```diff
@@ -104,7 +105,8 @@ type EndpointChangeTracker struct {
 	// items maps a service to is endpointsChange.
 	items map[types.NamespacedName]*endpointsChange
 	// makeEndpointInfo allows proxier to inject customized information when processing endpoint.
-	makeEndpointInfo makeEndpointFunc
+	makeEndpointInfo     makeEndpointFunc
+	onEndpointsMapChange endpointsMapChangeFunc
```

Comment:
Add a comment describing this function. Describing when it is used, expectation and etc. 

I would imagine the the function must not mutate the input maps.